exercise-1.js

```diff
@@ -1,143 +1,288 @@
 
         //example : [10,23,3,4] => function() => {10 : 'even', 23 : 'odd', 3 : 'odd', 4 : 'even'}
 
         //your code...
+    {
+        function evenOrOdd(myArr){
+            const myMap = {};
+            for(let i = 0; i < myArr.length; i++){
+              if(myArr[i] % 2 === 0){
+                  console.log(myArr[i] + ' : even')
+                  myMap[myArr[i]] = "even";
+                  }else{
+                    console.log(myArr[i] + ' : odd')
+                    myMap[myArr[i]] = "odd";
+                  }
+              }
+              return myMap;
+          }
+        evenOrOdd([4, 9, 4, 23, 99, 0, 100000000099])
+    }
 
 
 
 
 
-
-
+        //9 problem
         /************************************************************************************/
         const numbers = [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ];
         // Write a "for" loop that console.log()'s the first value in the  array, and every 3rd number, i.e. 0, 3, 6, 9: the zeroth, third, sixth, and ninth values.
 
         //your code...
+    {
+        for(let i = 0; i < numbers.length; i % 3 === 0 ? i += 3 : i++){
+            console.log(i)
+        }
+    }
 
 
 
+        //10 problem
         /************************************************************************************/
+    {
         const foodArray = [ 'potatoes', 'tamales', 'lemon','strawberries','chocolate', 'pudding', {program : 'TEKcamp'} ];
         //access the value of the last element of the array and set it to a variable called school.  print the school variable to the console.
+        //your code
+
+        let retrieve = foodArray[foodArray.length-1];
+        let school = retrieve.program;
 
         const adjectiveArray = [ 'salty', 'spicy', 'sour', 'sweet', 'rich','creamy','amazing'];
-        // Using both the foodArray and the adjectiveArray, write "for" loop that console.log()'s a sentence for each corresponding value in the arrays. Add the word "is" or "are" depending on if the food is singular or plural.  i.e. "Potatoes are salty", "Lemon is sour".
+        // Using both the foodArray and the adjectiveArray, write "for" loop that console.log()'s a sentence for each corresponding value in the arrays. Add the word "is" or "are" depending on if the food is singular or plural.  i.e. "Potatoes are salty", "Lemon is sour"
 
+        //your code
 
-        /************************************************************* */
-        // Refactor the for() loop to be a while loop.
+        for(let i = 0; i < foodArray.length; i++){
+            let word = foodArray[i];
+            if(word === retrieve){
+            console.log(school + ' is ' + `${adjectiveArray[i]}`)
+          }
+          else if((word.substring(word.length - 1) == 's')){
+            console.log(word + ' are ' + `${adjectiveArray[i]}`)
+          }else{
+              console.log(word + ' is ' + `${adjectiveArray[i]}`)
+            }
+        }
```

Comment:
watch your indentations here in the for loop.  everything that is indented should be indented one additional level deep.

Reply:
In general, having too many if / else checks in a for loop gets to be a little difficult to reason about.  

here is another solution, keeping in mind the concept of functions doing one thing. 

```

function combineArrayMessages(arr1,arr2) {
            
            const combinedArr = [];
            
            // filter out non-primitives from the input arrays
            const [primitives1,primitives2] = removeNonPrimitives(arr1,arr2);
            
            primitives1.forEach( (n,i) => {
                const msg = mapPluralToMsg(arr1[i]); // set the message to 'is' or 'are' depending on if the word is plural.
                combinedArr[i] = primitives1[i] + msg +  primitives2[i]; //combine both arrays together with the message.  
            })

            return combinedArr;
        }

        const mapPluralToMsg = word => {
            const last = word[word.length-1];
            return last === 's' ? " are " : " is ";
        }

        const removeNonPrimitives = (...arrays) => {
            const filteredArrays = [];
            for(arr of arrays) {
                const primitives = arr.map(el =>  typeof el === 'object' ? getFirstObjValue(el) : el);
                filteredArrays.push(primitives);
            }
            return filteredArrays;
        }

        const getFirstObjValue = obj => {
            const key = Object.keys(obj)[0];
            console.log(obj, key);
            return obj[key];
        }
        


        combineArrayMessages(foodArray,adjectiveArray);`

Reply:
You can decompose the solution I wrote even further, but just to give you an idea.